internal/datacoord/compaction_trigger.go

```diff
@@ -683,15 +670,27 @@ func (t *compactionTrigger) handleClusteringCompactionSignal(signal *compactionS
 			}
 		}
 
-		plans := t.clusteringCompactionManager.fillClusteringCompactionPlans(group.segments, clusteringKeyField.FieldID, ct)
+		clusteringCompactionTask := &datapb.CompactionTask{
+			// PlanId:
+			TriggerId:          signal.id,
+			State:              datapb.CompactionTaskState_init,
+			StartTime:          ts,
+			Type:               datapb.CompactionType_ClusteringCompaction,
+			CollectionTtl:      ct.collectionTTL.Nanoseconds(),
+			CollectionId:       signal.collectionID,
+			PartitionId:        group.partitionID,
+			Channel:            group.channelName,
+			ClusteringKeyField: clusteringKeyField,
+		}
+		t.clusteringCompactionManager.fillClusteringCompactionTask(clusteringCompactionTask, group.segments)
 		// mark all segments prepare for clustering compaction
-		t.setSegmentsCompacting(plans, true)
-		for _, plan := range plans {
-			clusteringCompactionJob.addCompactionPlan(plan, pipelining)
+		for _, seg := range group.segments {
+			t.meta.SetSegmentCompacting(seg.ID, true)
```

Comment:
should call t.meta.SetSegmentCompaction(seg.ID, false) if compaction task failed or completed?